hathor/indexes/memory_address_index.py

```diff
@@ -52,26 +53,16 @@ def subscribe_pubsub_events(self) -> None:
         # Subscribe to voided/winner events
         self.pubsub.subscribe(HathorEvents.CONSENSUS_TX_UPDATE, self.handle_tx_event)
 
+    def _extract_keys(self, tx: BaseTransaction) -> Iterable[str]:
+        return tx.get_related_addresses()
+
     def add_tx(self, tx: BaseTransaction) -> None:
         """ Add tx inputs and outputs to the wallet index (indexed by its addresses).
         """
-        assert tx.hash is not None
-
+        super().add_tx(tx)
         addresses = tx.get_related_addresses()
```

Comment:
Maybe use `self. _extract_keys`?

Reply:
I refactored this part and removed this call.